ecosystem/indexer/src/indexer/fetcher.rs

```diff
@@ -44,73 +45,33 @@ impl TransactionFetcherTrait for TransactionFetcher {
-    async fn fetch_next(&mut self) -> Transaction {
-        let mut transactions_buffer = self.transactions_buffer.lock().await;
-        if transactions_buffer.is_empty() {
-            // Fill it up!
-            loop {
-                let res = self
-                    .client
-                    .get_transactions(Some(self.version), Some(TRANSACTION_FETCH_BATCH_SIZE))
-                    .await;
-                match res {
-                    Ok(response) => {
-                        FETCHED_TRANSACTION.inc();
-                        let mut transactions = response.into_inner();
-                        transactions.reverse();
-                        *transactions_buffer = transactions;
-                        break;
-                    }
-                    Err(err) => {
-                        let err_str = err.to_string();
-                        // If it's a 404, then we're all caught up; no need to increment the `UNABLE_TO_FETCH_TRANSACTION` counter
-                        if err_str.contains("404") {
-                            aptos_logger::debug!(
-                            "Could not fetch {} transactions starting at {}: all caught up. Will check again in {}ms.",
-                            TRANSACTION_FETCH_BATCH_SIZE,
-                            self.version,
-                            RETRY_TIME_MILLIS,
-                        );
-                            tokio::time::sleep(Duration::from_millis(RETRY_TIME_MILLIS)).await;
-                            continue;
-                        }
-                        UNABLE_TO_FETCH_TRANSACTION.inc();
-                        aptos_logger::error!(
-                            "Could not fetch {} transactions starting at {}, will retry in {}ms. Err: {:?}",
-                            TRANSACTION_FETCH_BATCH_SIZE,
-                            self.version,
-                            RETRY_TIME_MILLIS,
-                            err
-                        );
-                        tokio::time::sleep(Duration::from_millis(RETRY_TIME_MILLIS)).await;
-                    }
-                };
-            }
-        }
-        // At this point we're guaranteed to have something in the buffer
-        let transaction = transactions_buffer.pop().unwrap();
-        self.version += 1;
-        transaction
-    }
-
-    /// fetches one version; this used for error checking/repair/etc
-    /// In the event it can't, it will keep retrying every RETRY_TIME_MILLIS ms
-    async fn fetch_version(&self, version: u64) -> Transaction {
+    /// Fetches all transactions within a block. If block height is not available, set block height using
+    /// the starting version
+    async fn fetch_block(&mut self) -> Vec<Transaction> {
         loop {
-            let res = self.client.get_transaction_by_version(version).await;
+            let res = self.client.get_block(self.block_height, true).await;
```

Comment:
This is going to make a LOT more network requests, no?

Reply:
Why would it be more? It would be more for empty blocks but the issue we're running into is due to big blocks right? So this should be more performant? It would be cool to be able to query multiple blocks based on block size.